notifier/notify_windows.go

```diff
@@ -1,16 +1,394 @@
+		log.Errorf("failed adding appID when building SnoreToast-Close-Command: %s Notification: %+v", err, n)
+		return
+	}
+
+	snoreToastRunCmdAsync(exec.Command(notifierPath, args...))
+}
+
+// internal functions:
+
+func initNotifierPath() {
+	notifierPathMutex.Lock()
+	defer notifierPathMutex.Unlock()
+
+	if notifierPath == "" {
+		var err error
+		notifierPath, err = getPath("notifier-snoretoast")
+
+		if err != nil {
+			log.Errorf("failed obtaining SnoreToast-Path: %s %s", err, notifierPath)
+			return
+		}
+	}
+}
+
+func initLogo() {
+	if _, err := os.Stat(logoLocation); err != nil { // File doesn't exist or another Error that is handled while copying file there
+		if err = os.MkdirAll(filepath.Dir(logoLocation), 0644); err != nil {
+			log.Errorf("failed to create Directory %s for Logo for SnoreToast: %s", logoLocation, err)
+		}
+		if err = downloadFile(logoLocation, webLogoLocation); err != nil {
+			log.Errorf("failed to copy Logo for SnoreToast from %s to %s: %s", webLogoLocation, logoLocation, err)
+		}
+	}
+}
+
+func notificationListener() {
+	log.Debugf("starting Callback-Pipe for SnoreToast: %s", pipeName)
+	l, err := winio.ListenPipe(pipeName, nil)
+	if err != nil {
+		log.Errorf("failed to start Pipe for SnoreToast: %s", err)
+		return
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			log.Errorf("failed to accept namedPipe-connection (for receiving Callbacks from SnoreToast): %s", err)
+			continue
+		}
+		go handlePipeMessage(conn)
+	}
+}
+
+func handlePipeMessage(conn net.Conn) {
+	data := snoreToastStrToStruct(readWstrConnToStr(conn))
+	log.Debugf("handling PipeMessage: %+v", data)
+
+	if data.action == "timedout" {
+		return
+	}
+
+	notificationsLock.Lock()
+	var n *Notification
+	for _, elem := range notifications {
+		if elem.GUID == data.id {
+			n = elem
+			break
+		}
+	}
+	notificationsLock.Unlock()
+	if n == nil {
+		return
+	}
+	n.Lock()
+	defer n.Unlock()
+
+	switch data.action {
+	case "dismissed": // do nothing
+	case "clicked": // like dismissed: do nothing
+	case "buttonClicked":
+		for _, actions := range n.AvailableActions {
+			if actions.Text == data.button {
+				n.SelectAction(actions.ID)
+				return
+			}
+		}
+
+		log.Warningf("failed to handle button click: button %s is reported to have been clicked but is no registered button. Available options: %+v", data.button, n.AvailableActions)
+	default:
+		log.Debugf("failed to handle SnoreToast-Action %s: not implemented yet", data.action)
+	}
+}
+
+// Snoretoast helper-Functions:
+
+// Verifies that the Argument contians no Semicolon which would be difficult (in some cases impossible) to parse in a Pipe-Response
+func snoreToastVerifyArgumentSyntax(arg string) error {
+	if strings.Contains(arg, ";") {
+		return fmt.Errorf("the SnoreToast-Argument %s would contain a semicolon which would screw up the pipe responses", arg)
+	}
+	return nil
+}
+
+func snoreToastStrToStruct(in string) actionCallback {
```

Comment:
I suggest `parseSnoreToastActionCallback(message string) actionCallback` as function signature. Makes it easier to read.